frontend/test/resolution/testRuntimePrimitives.cpp

```diff
@@ -0,0 +1,352 @@
+static void test11() {
+  Context ctx;
+  auto context = &ctx;
+  QualifiedType qt =  resolveTypeOfXInit(context,
+                         R""""(
+                           var x = __primitive("get_union_id");
+                         )"""");
+  assert(qt.kind() == QualifiedType::CONST_VAR);
+  auto typePtr = qt.type();
+  assert(typePtr);
+  assert(typePtr->isIntType());
+  assert(typePtr->toIntType()->bitwidth() == 64);
+}
+
+static void voidTypeTestHelper(std::string program) {
+  Context ctx;
+  auto context = &ctx;
+  QualifiedType qt =  resolveTypeOfXInit(context, program);
+  assert(qt.kind() == QualifiedType::CONST_VAR);
+  auto typePtr = qt.type();
+  assert(typePtr);
+  assert(typePtr->isVoidType());
+}
+
+// test for primitive "chpl_init_record", which should return void
+static void test12() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_init_record");)""");
+}
+
+// test for primitive "chpl_comm_get", which should return void
+static void test13() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_get");)""");
+}
+
+// test for primitive "chpl_comm_put", which should return void
+static void test14() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_put");)""");
+}
+
+// test for primitive "chpl_comm_array_get", which should return void
+static void test15() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_array_get");)""");
+}
+
+// test for primitive "chpl_comm_array_put", which should return void
+static void test16() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_array_put");)""");
+}
+
+// test for primitive "chpl_comm_remote_prefetch", which should return void
+static void test17() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_remote_prefetch");)""");
+}
+
+// test for primitive "chpl_comm_get_strd", which should return void
+static void test18() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_get_strd");)""");
+}
+
+
+// test for primitive "chpl_comm_put_strd", which should return void
+static void test19() {
+  voidTypeTestHelper(R"""(var x = __primitive("chpl_comm_put_strd");)""");
+}
+
+// test for primitive "shift_base_pointer", which should return void
+static void test20() {
+  voidTypeTestHelper(R"""(var x = __primitive("shift_base_pointer");)""");
+}
+
+// test for primitive "array_set", which should return void
+static void test21() {
+  voidTypeTestHelper(R"""(var x = __primitive("array_set");)""");
+}
+
+// test for primitive "array_set_first", which should return void
+static void test22() {
+  voidTypeTestHelper(R"""(var x = __primitive("array_set_first");)""");
+}
+
+// test for primitive "auto destroy runtime type", which should return void
+static void test23() {
+  voidTypeTestHelper(R"""(var x = __primitive("auto destroy runtime type");)""");
+}
+
+// test for primitive "create fn type", which should return void
+static void test24() {
+  voidTypeTestHelper(R"""(var x = __primitive("create fn type");)""");
+}
```

Comment:
A nit, but I don't think it's worth making different functions for all of these; I'd make `static void testVoidPrims() {` and then have all the calls to `voidTypeTestHelper` you have here. But, it does not bother me if this merges without this change.

Reply:
I refactored it based on this